proto/redpanda/api/dataplane/v1alpha1/topic.proto

```diff
@@ -0,0 +1,261 @@
+      responses: {
+        key: "204"
+        value: {
+          description: "Topic was deleted successfully"
+          schema: {}
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+  }
+  rpc GetTopicConfiguration(GetTopicConfigurationRequest) returns (GetTopicConfigurationResponse) {
+    option (google.api.http) = {get: "/v1alpha1/topics/{topic_name}/configuration"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Get Topic Configuration"
+      description: "Get Topic Configuration."
+      responses: {
+        key: "200"
+        value: {
+          description: "Ok"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.GetTopicConfigurationResponse"}
+          }
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+  }
+  rpc UpdateTopicConfiguration(UpdateTopicConfigurationRequest) returns (UpdateTopicConfigurationResponse) {
+    option (google.api.http) = {
+      patch: "/v1alpha1/topics/{topic_name}/configuration"
+      body: "configuration"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Update Topic Configuration"
+      description: "Updates a subset of the topic configuration."
+      responses: {
+        key: "200"
+        value: {
+          description: "Ok"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.UpdateTopicConfigurationResponse"}
+          }
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+  }
+  rpc SetTopicConfiguration(SetTopicConfigurationRequest) returns (SetTopicConfigurationResponse) {
+    option (google.api.http) = {
+      put: "/v1alpha1/topics/{topic_name}/configuration"
+      body: "configuration"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Set Topic Configuration"
+      description: "Set the entire configuration of a topic. Config entries not provided in the request are removed."
+      responses: {
+        key: "200"
+        value: {
+          description: "Ok"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.SetTopicConfigurationResponse"}
+          }
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+  }
+}
```

Comment:
Should there be some validation for messages in this file?

Reply:
Same as above, the protos for anything but user.proto are just drafts for the OpenAPI spec.